### DISCUSSION TITLE
xDrip+ event logs and screenshots thereof

### DISCUSSION THREAD

**Discussion by jbowler** [NightscoutFoundation/xDrip#2794] [Ideas]
This discussion originates in this, FaceBook, discussion:

https://www.facebook.com/groups/xDripG5/posts/9041152629291485/?comment_id=9041252579281490&reply_comment_id=9056028927803855

My original point is that \<three drops\>/View Events Log gets overloaded with spurious warnings.  In the case of the FB thread this seemed to be NightScout warnings which were clearly not directly related to the issue the OP (in that thread) posted.

I take Navid's point that the developers have the "upload logs" option, which doesn't truncate on screen size, and that the developers have History to deal with.

So my point, and my reason to create a discussion here, is that it is counter productive to include /ancient history/ or, indeed, /really important to me/ in event logs that FB users will be asked to post.  If the log in question contained NOTHING it would be more useful; so start from there.

Only put something in a bug report that will be reported by people encountering bugs that will certainly be relevant, and work up to the screen size.

**Comment by Navid200** [comment]
This shows the screenshot that was posted on facebook in the thread:
![341763485_607935074325452_8918289884371249696_n](https://user-images.githubusercontent.com/51497406/233744381-16e4af53-1007-4483-9068-de60984f592a.jpg)

**Comment by jbowler** [comment]
Yes.  So the problem is that the OP originally reported a three minute failure from 12:56 (IRC; I'm quoting from memory and the screen shots don't seem to contain the calendar date even).  BTW; I'm not implying any miscreance in the OP comment.  The data as presented seems 100% correct.

The sensor was, so far as I can tell, inserted and the transmitter too, then nothing happened, so far as anyone knows.

The whole issue raise by the original FB thread is off topic here; my intention was to point out that the excess data in the screen shot obfuscates any possible attempt to work out what is going on.

**Comment by Navid200** [comment]
This is my opinion:

The System status page is crucial.
The guide is very important.
The logs present information.

In case of a G6 not establishing connectivity, as can be seen in the status page below from the same thread, there are so many questions that must be answered before anyone may need to see the logs.
![342052326_1160258197975273_2260970130283484191_n](https://user-images.githubusercontent.com/51497406/233748532-fef2e392-6988-47a9-a4f5-cf0ec99c41b8.jpg)

For example:
Is the transmitter id entered correctly (possible typo)?
Is there another app already connected to the same transmitter?
Is the transmitter activated?  If it's still in the box, it is not.
Is the transmitter transmitting?  You can use a scanner to verify.
Are the settings correct?
....

But, the user was not very interested.  It seemed she had already decided that something was wrong with the particular phone she was using.

**Comment by jamorham** [comment]
I'm not sure what the problem being reported here is exactly, but the log shows messages about receiving remote BG when it isn't set as a follower. Isn't that a big clue that something is going on with that?

**Comment by Navid200** [comment]
@jbowler I just realized this is an issue you have opened.
I don't know why you did that.
I am certain you saw a note asking you not to open an issue unless you were told by a developer.

**Comment by jbowler** [comment]
Well, I opened it from here:

https://github.com/NightscoutFoundation/xDrip/discussions/1900

It's meant to be a  **discussion**; read the first line:

>This discussion originates in this, FaceBook, discussion:

So can you move it to the "Discussions" (I've found out how to open one, but it's more efficient to move this "Issue" since it was meant to  be a "Discussion" in the first place...)

**Comment by Navid200** [comment]
@jbowler Please do NOT open an issue.
Please open a discussion here: https://github.com/NightscoutFoundation/xDrip/discussions

If needed, we will convert it to an issue.

You must have seen this same request when you opened the issue.  It was a request from you as an xDrip user by those who maintain it.
You can work with us or against us.  Why would you choose the latter?  How will that serve you?

**Reply by jbowler** [reply]
Thanks for converting it.  I certainly intended to start a discussion, not an issue.  It might be worth disabling the "Create a New Issue from a Discussion" button on #1900 if this is possible since that is where anyone new to discussions is likely to end up first.

**Comment by jbowler** [comment]
After some experiment I have a few suggestions.  The NightScout messages that are flooding the event log seem to be "Mid".  Turning "Mid" off seems to remove them.  I have other examples of messages flooding the log (also, it happens, about NightScout REST-API stuff).  There are obvious ways of handling specific messages but a general solution is to initially turn off "Low" and "Mid"; I'm assuming the checkboxes only affect the screen display (so the screenshots) not the "Upload Logs" result.

I got much more satisfactory results, more useful to me, by doing this:

![Error Log Settings](https://user-images.githubusercontent.com/512616/233801516-56ece2ba-137b-4bc7-ac60-11b063b45b09.jpg)

As you can see both the two left hand images (with Low+Mid and Mid on) are entirely dominated by 5-minute logs.  It is true that the logs are all NightScout related and it is also true that in the past, before I was obliged to use NightScout, I was able to divine useful information for debugging Dexcom G6 settings from the information that was on page 1 of the screen.

So suggestion (1) is to disable Low _and_ Mid by default.

Suggestion (2) is to disable everything and to replace the blank screen that results with more detailed instructions about what the checkboxes do; the names are not very helpful and probably not meaningful for most people.  This is obviously more work.

Suggestion (3) is to do what, IRC, the Linux Kernel does; replace repeated messages with a count and a line of the general form, "the last message was repeated 1024 times".  That may or may not be easy depending on who y'all are identifying the messages but if i18n support is there that tends to help a lot.  This suggestion helps address the kind of mess of messages here:

![Repeated remote BG packet error](https://user-images.githubusercontent.com/512616/233802231-65c76daf-6bbb-4e41-8794-630134510331.jpg) ![Error log, repeated DNS lookup failure](https://user-images.githubusercontent.com/512616/233802237-246e9af8-4ce5-4b08-8623-f8922632e62b.jpg)

In both cases the same "Mid" message is repeated many times and in both cases there are no other messages being output; you will have to believe me with the first image, I used the Mid setting to locate the messages more rapidly in the log.  Repeating the message doesn't help, particularly the second one which is so long.  Yes, the first really is a configuration error (I was trying to backfill CGM data from NightScout) and the second really is a bug in either xDrip+ or Android (it happened after a routing change), but my suggestion is to stop repeating the message at the very least in the cases where it is exactly the same (apart from the timestamp) as the previous one.

Obviously there are other things which _could_ be done, but the possibilities seem to me to be a lot of work for gain that might only be temporary, e.g. shortening specific messages or dropping the repeats in the code that generates them rather than suggestion (3) having the error logging do it.

**Reply by jamorham** [reply]
What is your configuration that you are receiving BG data intended for follower when your handset is not set for this? What is your data source?

**Reply by jbowler** [reply]
> What is your configuration that you are receiving BG data intended for follower when your handset is not set for this? What is your data source?

That's very much a separate topic and, in fact, I don't know the answer for sure (you will note that the errors come from two days ago.)  I was setting up my new 'phone (a OnePlus 10T).  I had attempted to clone the old 'phone (a Samsung S10) using the QR code.  So the old 'phone was set up and running with a transmitter sensor.  Unlike the FB OP I wanted to verify the new phone without sacrificing a new sensor and, certainly, I couldn't afford to activate a new transmitter (I was only 19 days in.)  I had installed the xDrip+ nightly build per the Android APS instructions (IRC before I started using AAPS I'd always used a stable build).

I had first linked xDrip+ up to NightScout in an attempt to get BG data from there; the old phone xDrip+ was feeding CGM into NightScout.  Anyway, I messed around with the settings a lot to get data that I could use for testing AndroidAPS; the latter is where I expected to any problems.  Eventually I gave up on that; I'm not sure why I didn't just try using xDrip+ as a follower, I just didn't think of doing that.

What I did do, however, is to swap the new phone xDrip+ to use slot 1; this gave me a much better test environment because I was then able to connect to the transmitter and "start" the old sensor.  I did this at 20230419 0836  This worked great, but at that point the system status "Followers" screen was displaying this:

![20230420 Master;Follower issue](https://user-images.githubusercontent.com/512616/233807140-92212c94-99ef-4341-8ccc-d97878e0f95a.jpg)

That's on the OnePlus 10T.  Amusingly the Samsung S10, "John's S10" was displaying the Dr Who message too; "We are the Master" but declaiming that the Master was the _new_ phone, the OnePlus!

Some time after doing this the old sensor expired and I activated a new one from the new 'phone.  xDrip+ history shows that I did that at 20230420 13:54; shortly **_after_** the messages I posted previously.

So you can see the confusion; I had "Download treatments" turned on in settings/Cloud Update/NightScout and I think that must be where the BG values were coming from but I had also managed to create a really weird setup.  The FB OPs setup is likely to be a more productive place to look since I doubt the OP had enabled the "Manual Slot" option for the G6.

Nevertheless I did download the complete log for you as a result of the "ActivityRecognizer" report.  This is not the one at 17:12 today that you can see in the fourth screen above, it was an earlier one at 2023-04-19 11:31:54, so about 3 hours after I "started" the **old** sensor on the OnePlus 10T.
